h2o-algos/src/main/java/hex/rulefit/RuleFit.java

```diff
@@ -90,6 +90,11 @@ public void init(boolean expensive) {
         // if (expensive && error_count() == 0) checkMemoryFootPrint();
     }
 
+    @Override
+    protected void ignoreBadColumns(int npredictors, boolean expensive) {
+        // Do not remove bad columns - we need them for correct GLM train frame creation
```

Comment:
this does not seem right, I think you might want to remove bad/constant cols at the beginning of RuleFit

Reply:
I think the fact that this fixes an issue might be a sign of something bad happening

@koniecsveta how are frames passed to glm and drf/gbm? do you use trainModelNested?

Reply:
no no I don't use trainModelNested and removing bad cols at the beginning didn't help

Reply:
like this, it still goes into ModelBuilder#ignoreBadCols() on glm's and drf/gbm's model builder level, by this I just avoid going there on rulefit's model builder level